packages/react-components/react-combobox/src/utils/useComboboxBaseState.ts

```diff
@@ -172,7 +173,16 @@ export const useComboboxBaseState = (
       }
     }
     // this should only be run in response to changes in the open state or children
-  }, [open, children, activeDescendantController]);
+  }, [open, children, activeDescendantController, getOptionById]);
+
+  const onActiveDescendantChange = React.useCallback(
```

Comment:
```suggestion
  const onActiveDescendantChange = useEventCallback(
```

This should keep the callback completely stable so that you avoid creating the event listener multiple times